sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/WebappController.java

```diff
@@ -0,0 +1,203 @@
+package io.javaoperatorsdk.operator.sample;
+
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.ExecListener;
+import io.fabric8.kubernetes.client.dsl.ExecWatch;
+import io.javaoperatorsdk.operator.api.*;
+import io.javaoperatorsdk.operator.processing.event.EventSourceManager;
+import io.javaoperatorsdk.operator.processing.event.internal.CustomResourceEventSource;
+import io.javaoperatorsdk.operator.processing.event.internal.InformerEventSource;
+
+import okhttp3.Response;
+
+import static io.javaoperatorsdk.operator.processing.event.DefaultEventSourceManager.CUSTOM_RESOURCE_EVENT_SOURCE_NAME;
+
+@Controller
+public class WebappController implements ResourceController<Webapp> {
+
+  private KubernetesClient kubernetesClient;
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  public WebappController(KubernetesClient kubernetesClient) {
+    this.kubernetesClient = kubernetesClient;
+  }
+
+  @Override
+  public void init(EventSourceManager eventSourceManager) {
+    InformerEventSource<Tomcat> tomcatEventSource =
+        new InformerEventSource<>(kubernetesClient, Tomcat.class, t -> {
+          var cache =
```

Comment:
What is this doing here?

Reply:
added explanation in a comment:

To create an event to a related WebApp resource and trigger the reconciliation
we need to find which WebApp this Tomcat custom resource is related to.
To find the related UID of the WebApp resource we traverse the cache to
and identify it based on naming convention.

As a remember this was a api call before, with cache this is more efficient, in v2 we address the resources differently there this will be trivial without the cache. But we are looking now into how to make this able to do efficiently in general.